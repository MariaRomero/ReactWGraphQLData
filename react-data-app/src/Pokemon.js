import React from "react";

const detail = {
  padding: "0 6px"
};

const Pokemon = ({
  item: { name, number, types, resistant, image, weaknesses }
}) => {
  return (
    <>
      <div>
        <img
          className="Pokemon"
          alt={name}
          src={image}
          style={{
            height: "150px",
            padding: "10px 10px 10px 10px",
            width: "150px",
            transform: "translate(-50%, 0)",
            marginLeft: "50%"
          }}
        />
        <p style={detail}>Name: {name}</p>
        <p style={detail}>Number: {number}</p>
        <p style={detail}>Types: {types.join(", ")}</p>
        <p style={detail}>Resistant: {resistant.join(", ")}</p>
        <p style={detail}>Weaknesses: {weaknesses.join(", ")}</p>
      </div>
    </>
  );
};

export default Pokemon;
