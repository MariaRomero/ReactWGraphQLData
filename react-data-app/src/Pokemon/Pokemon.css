.pokemon-img {
  height: 150px;
  padding: 10px 10px;
  width: 150px;
  transform: translate(-50%, 0);
  margin-left: 50%;
}

.poke-text {
  padding: 0 6px;
}
