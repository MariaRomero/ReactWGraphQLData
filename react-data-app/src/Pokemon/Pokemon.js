import React from "react";
import "./Pokemon.css";

const Pokemon = ({
  item: { name, number, types, resistant, image, weaknesses }
}) => {
  return (
    <>
      <div>
        <img className="pokemon-img" alt={name} src={image} />
        <p className="poke-text">Name: {name}</p>
        <p className="poke-text">Number: {number}</p>
        <p className="poke-text">Types: {types.join(", ")}</p>
        <p className="poke-text">Resistant: {resistant.join(", ")}</p>
        <p className="poke-text">Weaknesses: {weaknesses.join(", ")}</p>
      </div>
    </>
  );
};

export default Pokemon;
