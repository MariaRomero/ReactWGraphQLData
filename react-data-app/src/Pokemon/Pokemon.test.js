import React from "react";
import Pokemon from "./Pokemon";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

test("Renders Pokemon when props are passed", () => {
  const { container } = render(
    <Pokemon
      item={{
        name: "Pikachu",
        number: "001",
        types: ["test"],
        resistant: ["blah"],
        image: "hhtp://123",
        weaknesses: ["Hello", "World"]
      }}
    />
  );

  expect(container.firstChild).toMatchSnapshot();
});
