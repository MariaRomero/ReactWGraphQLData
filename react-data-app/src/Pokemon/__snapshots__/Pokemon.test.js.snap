// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Pokemon when props are passed 1`] = `
<div>
  <img
    alt="Pikachu"
    class="pokemon-img"
    src="hhtp://123"
  />
  <p
    class="poke-text"
  >
    Name: 
    Pikachu
  </p>
  <p
    class="poke-text"
  >
    Number: 
    001
  </p>
  <p
    class="poke-text"
  >
    Types: 
    test
  </p>
  <p
    class="poke-text"
  >
    Resistant: 
    blah
  </p>
  <p
    class="poke-text"
  >
    Weaknesses: 
    Hello, World
  </p>
</div>
`;
